datasets/xtreme_s/xtreme_s.py

```diff
@@ -0,0 +1,780 @@
+            yield from self._minds14_generate_examples(*args, **kwargs)
+        elif self.config.dataset_name == "babel":
+            yield from self._babel_generate_examples(*args, **kwargs)
+
+    # MLS
+    def _mls_split_generators(self, dl_manager):
+        lang = _ID_TO_LANG[self.config.lang_name]
+
+        archive_path = dl_manager.download_and_extract(self.config.data_urls[0].format(lang))
+        data_path = os.path.join(archive_path, f"mls_{_ID_TO_LANG[self.config.lang_name]}")
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "data_dir": os.path.join(data_path, "train"),
+                    "sub_folder": "limited_supervision/9hr",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"data_dir": os.path.join(data_path, "dev")},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"data_dir": os.path.join(data_path, "test")},
+            ),
+        ]
+
+    def _mls_generate_examples(self, data_dir, sub_folder=""):
+        """Generate examples from a Multilingual LibriSpeech data dir."""
+        transcript_path = os.path.join(data_dir, "transcripts.txt")
+        key = 0
+        all_ids = None
+
+        # find relevant ids
+        sub_path = os.path.join(data_dir, sub_folder)
+        all_ids_paths = glob.glob(sub_path + "/*/*.txt") + glob.glob(sub_path + "/*.txt")
+        all_ids = []
+        if sub_folder != "":
+            for path in all_ids_paths:
+                with open(path, "r", encoding="utf-8") as f:
+                    all_ids += [line.strip() for line in f.readlines()]
+
+            all_ids = set(all_ids)
+
+        with open(transcript_path, "r", encoding="utf-8") as f:
+            for line in f:
+                line = line.strip()
+                _id, transcript = line.split("\t")
+
+                if sub_folder != "" and _id not in all_ids:
+                    # filter-out audios not contained in the 9/10h version
+                    continue
+
+                audio_file = f"{_id}.flac"
+                speaker_id, chapter_id = [int(el) for el in _id.split("_")[:2]]
+
+                yield key, {
+                    "path": os.path.join(data_dir, "audio", str(speaker_id), str(chapter_id), audio_file),
+                    "audio": os.path.join(data_dir, "audio", str(speaker_id), str(chapter_id), audio_file),
+                    "transcription": transcript,
+                }
+                key += 1
+
+    # Voxpopuli
+    def _voxpopuli_split_generators(self, dl_manager):
+        # TODO(PVP) - need to add warning here that even for a single
+        # language ALL files need to be downloaded because all langs archive_path
+        # mixed in single audio files
+        if hasattr(dl_manager, "download_config"):
```

Comment:
why do you need to check this with `hasattr` ?

Reply:
copy-paste from other datasets :sweat_smile: so I guess I don't need it haha

Reply:
yes you can remove it :)